### PR TITLE
MGMT-18380: Update package naming in generator_test.go for black-box testing methodology

### PR DESCRIPTION
**Pull request by linoyaslan**
This PR updates the package naming in `generator_test.go` to use the "_test" suffix. This change follows the convention for organizing test files, making our project structure clearer. By using the black-box testing methodology, we focus on testing the package's functionality from an external perspective. Tests interact only with the package's exported interfaces and behavior. This simplifies our test suite, making it easier to add new tests and make changes in the future.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [X] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [X] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [X] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: pkg/staticnetworkconfig/generator_test.go [hunk 1]
```diff
@@ -21,272 +22,255 @@ func TestStaticNetworkConfig(t *testing.T) {
...
-  state: up
-  type: ethernet`
-		config, err := staticNetworkGenerator.generateConfiguration(hostYaml)
+		var (
+			hostsYAMLS = `[
+		{
+			"network_yaml": "%s"
+		}
+	]`
+			hostYaml = `interfaces:\n- ipv4:\n    address:\n      - ip: 192.0.2.1\n        prefix-length: 24\n    dhcp: false\n    enabled: true\n  name: eth0\n  state: up\n  type: ethernet`
```
Comment (rccrdpccl) [inline pkg/staticnetworkconfig/generator_test.go:54]: Could this be in a more readable format?

**Comment by rccrdpccl** [comment]
Very nice, probably the difficult thing will be applying this approach to the rest of codebase

**Comment by linoyaslan** [comment]
> Very nice, probably the difficult thing will be applying this approach to the rest of codebase

@rccrdpccl I agree. This ticket is part of the "Use nmstate.service instead of deploying NetworkManager keyfile" epic, aiming to shift the validations of static configuration from validate keyfiles to user-provided YAML. Test fixes were planned (which I've completed). However, due to project concerns, the validation fixes are on hold.

**Comment by AlonaKaplan** [comment]
> Very nice, probably the difficult thing will be applying this approach to the rest of codebase

I believe we should at least try to apply this approach on new tests we add/review.

File: pkg/staticnetworkconfig/generator_test.go (hunk 1 above)
Comment (linoyaslan) [reply pkg/staticnetworkconfig/generator_test.go:54]: @rccrdpccl AFAIK,  YAML content must be properly escaped to fit within a JSON string. Newlines in hostYAML can disrupt the JSON structure. Using a string literal with \n for newlines simplifies this by avoiding manual escaping and string manipulation. I didn't find any other simple alternatives.

**Comment by rccrdpccl** [comment]
> I believe we should at least try to apply this approach on new tests we add/review.

I agree, hopefully more people will be onboard with this

File: pkg/staticnetworkconfig/generator_test.go (hunk 1 above)
Comment (rccrdpccl) [reply pkg/staticnetworkconfig/generator_test.go:54]: Can we programmatically escape them? It's a minor concern, but if we could I think it'd make the test more readable.

Comment (linoyaslan) [reply pkg/staticnetworkconfig/generator_test.go]: @rccrdpccl I tried something and pushed the changes. What do you think?

File: pkg/staticnetworkconfig/generator_test.go [hunk 2]
```diff
@@ -21,272 +23,266 @@ func TestStaticNetworkConfig(t *testing.T) {
...
   state: up
-  type: ethernet`
-		config, err := staticNetworkGenerator.generateConfiguration(hostYaml)
+  ipv4:
+    enabled: true
+    address:
+      - ip: 192.0.2.1
+        prefix-length: 24`
+
+			escapedYamlContent = escapeYAMLForJSON(hostYAML)
```
Comment (linoyaslan) [inline pkg/staticnetworkconfig/generator_test.go:65]: @rccrdpccl  what do you think?

Comment (rccrdpccl) [reply pkg/staticnetworkconfig/generator_test.go:65]: I think it's more readable, nice :)

File: pkg/staticnetworkconfig/generator_test.go [hunk 3]
```diff
@@ -416,17 +459,24 @@ var _ = Describe("StaticNetworkConfig.GenerateStaticNetworkConfigArchive", func(
...
-func checkArchiveString(archiveString string, allData []StaticNetworkConfigData) {
+func checkArchiveString(archiveString string, allData []snc.StaticNetworkConfigData) {
 	for _, data := range allData {
 		Expect(archiveString).To(ContainSubstring("tar"))
 		Expect(archiveString).To(ContainSubstring(filepath.Join("/etc/assisted/network", data.FilePath)))
 		Expect(archiveString).To(ContainSubstring(data.FileContents))
 	}
 }
+
+// escapeYAMLForJSON takes a YAML content string and escapes necessary characters to ensure it can be safely embedded within a JSON string.
```
Comment (ori-amizur) [inline pkg/staticnetworkconfig/generator_test.go:477]: Why not use `json.Marshal` to do the job ? Example:
```suggestion
func escapeStringForJSON(s string) (string, error) {
	// Use json.Marshal to escape the string
	escaped, err := json.Marshal(s)
	if err != nil {
		return "", err
	}

	// json.Marshal returns a byte slice with double quotes around the string,
	// so we need to trim the double quotes
	return string(escaped[1 : len(escaped)-1]), nil
}
```

Comment (linoyaslan) [reply pkg/staticnetworkconfig/generator_test.go:477]: @ori-amizur As we discussed, this PR was merged just before I came across this comment, so I'll incorporate this change in another PR.

**Pull request merged**